scripts/zones/Arrapago_Reef/mobs/Medusa.lua

```diff
@@ -1,28 +1,45 @@
 -----------------------------------
 -- Area: Arrapago Reef
 -- NPC:  Medusa
+-- TODO: resists, attack/def boosts
 -----------------------------------
 
 require("scripts/globals/titles");
+require("scripts/zones/Arrapago_Reef/MobIDs");
+require("scripts/zones/Arrapago_Reef/TextIDs");
 
 -----------------------------------
 -- onMobSpawn Action
 -----------------------------------
 
 function onMobSpawn(mob)
+    mob:setLocalVar("eeshpp", math.random(5,99)); -- Uses EES randomly during the fight
 end;
 
 -----------------------------------
 -- onMobEngaged Action
 -----------------------------------
 
-function onMobEngaged(mob,target)
+function onMobEngaged(mob, target)
+    target:showText(mob, MEDUSA_ENGAGE);
+    SpawnMob(ExonOne,   180):updateEnmity(target);
+    SpawnMob(ExonTwo,   180):updateEnmity(target);
+    SpawnMob(ExonThree, 180):updateEnmity(target);
+    SpawnMob(ExonFour,  180):updateEnmity(target);
+end;
 
-	SpawnMob(16998863,180):updateEnmity(target);
-	SpawnMob(16998864,180):updateEnmity(target);
-	SpawnMob(16998865,180):updateEnmity(target);
-	SpawnMob(16998866,180):updateEnmity(target);
+-----------------------------------
+-- onMobFight Action
+-----------------------------------
 
+function onMobFight(mob, target)
+    local HPP = mob:getHPP();
+    if (mob:getLocalVar("usedees") == 0) then
+        if (HPP <= mob:getLocalVar("eeshpp")) then
+            mob:useMobAbility(479); -- Eagle Eye Shot

```

Comment:
@bendangelo  I could have done that scripted 2hour method, but I was not confident at all that it would use it. Plus, there is little/no documentation regarding this in the function header.  You asked me to submit the code I have. This method works just fine. The animation for EES works. I agree with @TeoTwawki with this being more accurate to retail 2hours, except that a mob should use their 2hour way before they hit 5% life. :)